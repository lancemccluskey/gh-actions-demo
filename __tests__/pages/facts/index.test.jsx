import { render, screen, fireEvent } from '@testing-library/react';
import Facts from '../../../pages/facts';

const testFact = "testfact";

describe('<Facts />', () => {
  it('renders', () => {
    const { container } = render(<Facts data={testFact} />);

    expect(container).toMatchSnapshot();
  });

  it('should display Not implemented if Like clicked', () => {
    render(<Facts data={testFact} />);

    expect(screen.queryByText('Not implemented')).toBeNull();
    const likeButton = screen.getAllByRole('button')[0];
    fireEvent.click(likeButton);

    expect(screen.getByText('Not implemented')).toBeInTheDocument();
  });

  it('should display new fact when clicking next', async () => {
    fetch.resetMocks();

    render(<Facts data={testFact} />);

    fetch.mockResponseOnce('newfact');

    const nextButton = screen.getAllByRole('button')[1];
    fireEvent.click(nextButton);

    await screen.findByText('newfact');
  });
});
