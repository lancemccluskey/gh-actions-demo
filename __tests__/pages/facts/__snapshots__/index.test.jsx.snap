// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Facts /> renders 1`] = `
<div>
  <div
    class="css-iiivav-Facts"
  >
    <div
      class="css-1nuvgh6-Card"
    >
      <span>
        testfact
      </span>
      <div
        class="css-13uxruf-Button"
      >
        <button
          class="css-uw5q0g-Button"
        >
          Like
        </button>
        <button
          class="css-uw5q0g-Button"
        >
          Next
        </button>
      </div>
    </div>
  </div>
</div>
`;
